src/utils/ProteomicsLFQ.cpp

```diff
@@ -1638,6 +1605,43 @@ class ProteomicsLFQ :
       }      
       design.setMSFileSection(msfs);
     }
+
+    // some sanity checks
+    // extract basenames from experimental design and input files
+    const auto& pl2fg = design.getPathLabelToFractionGroupMapping(true);      
+    set<String> ed_basenames;
+    for (const auto& p : pl2fg)
+    {
+      const String& filename = p.first.first;
+      ed_basenames.insert(filename);
+    }
+
+    set<String> in_basenames;
+    for (Size i = 0; i != in.size(); ++i)
+    {
+      const String& in_bn = File::basename(in[i]);
+      in_basenames.insert(in_bn);
+    }
+
+    if (!std::includes(ed_basenames.begin(), ed_basenames.end(), in_basenames.begin(), in_basenames.end()))
+    {
+      throw Exception::InvalidParameter(__FILE__, __LINE__,
+        OPENMS_PRETTY_FUNCTION, "Spectra file basenames provided as input need to match a subset the experimental design file basenames.");
+    }
+
+    design.filterByBasenames(in_basenames); 
```

Comment:
can we check if filtering needs to be performed and in that case output that only a subset is processed?

Reply:
The LOG INFO in the method is not enough?

Reply:
it is not that helpful if the user forgets to specify a file on the filename

Reply:
done by returning nr of filtered files